docs/stdlib/bool.rst

```diff
@@ -220,3 +222,114 @@ any(a) or any(b)``.
 
 For more customized handling of ``{}``, use the :eql:op:`?? <coalesce>`
 operator.
+
+
+----------
+
+
+.. eql:function:: std::assert( \
+                    input: bool, \
+                    named only message: optional str = <str>{} \
+                  ) -> bool
+
+    .. versionadded:: 3.0
+
+    Checks that the input bool is ``true``.
+
+    If the input bool is ``false``, ``assert`` raises a
+    ``QueryAssertionError``. Otherwise, this function returns ``true``.
+
+    .. code-block:: edgeql-repl
+
+        db> select assert(true);
+        {true}
+
+        db> select assert(false);
+        edgedb error: QueryAssertionError: assertion failed
+
+        db> select assert(false, message := 'value is not true');
+        edgedb error: QueryAssertionError: value is not true
+
+    ``assert`` can be used in triggers to create more powerful constraints. In
+    this schema, the ``Person`` type has both ``friends`` and ``enemies``
+    links. You may not want a ``Person`` to be both a friend and an enemy of
+    the same ``Person``. ``assert`` can be used inside a trigger to easily
+    prohibit this.
+
+    .. code-block:: sdl
+        
+          type Person {
+            required name: str;
+            multi friends: User;
+            multi enemies: User;
+
+            trigger prohibit_frenemies after insert, update for each do (
+              for obj in __new__ union (
```

Comment:
I'm not totally sure why I did it with a for loop in the commit-message example, but I think it would be better to skip the for loop and just use `__new__` directly in the trigger body.
(Putting a for loop in would be necessary if you are doing `for all`, but I don't think there's a good reason to do it that way.)